### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode character u'\xb0' in position 73: ordinal not in range(128)

### ISSUE THREAD

**Issue by coh7eiqu8thaBu** [viper-framework/viper#392]
When I try to open this EML file, i get this error :

```
malware viper > open -l 40
[*] Session opened on /home/user/.montage_crypt/Securite/viper/projects/malware/binaries/5/9/9/5/5995186cb71b7a82807985263e217443563aa9bcc1e6261234117581f8501914
Traceback (most recent call last):
  File "./viper.py", line 25, in <module>
    c.start()
  File "/home/user/.montage_crypt/Securite/viper/viper/core/ui/console.py", line 172, in start
    data = input(prompt).strip()
UnicodeEncodeError: 'ascii' codec can't encode character u'\xb0' in position 73: ordinal not in range(128)

```

and viper exit

impossible to push zip file ... so it's here : http://we.tl/tO0Sg7li4g

**Comment by Rafiot** [comment]
That's weird, I cannot reproduce the bug.

**Comment by botherder** [comment]
I suspect he had special characters in the zip filename.

**Comment by kevthehermit** [comment]
I was thinking we could do something similar to how yara module handles unicode. 

If string printable use string else replace with /x encoded 

something like this

```
        def string_printable(line):
            line = str(line)
            new_line = ''
            for c in line:
                if c in printstring.printable:
                    new_line += c
                else:
                    new_line += '\\x' + c.encode('hex')
            return new_line
```

anywhere you expect to find unicode pass the string to this function first.

**Comment by Rafiot** [comment]
I'd prefer printing unicode characters is possible

**Comment by deadbits** [comment]
Bottle and Python have similar issues with weird characters like this.

Something like this may work:
```
unicode(some_input, encoding='utf-8', errors='replace')
```

The input should still print as expected in a terminal after, as well being able to be displayed in Bottle without issue.

I only skimmed this issue so I may be a bit off on the full understanding of the problem here..

**Comment by Rafiot** [comment]
Should be fixed now.

**Comment by seamustuohy** [comment]
**DEAR DOCKER USERS FROM THE FUTURE: Here is what we figured out so far.[*](https://xkcd.com/979/)**

If you are using viper in a docker container you will encounter a similar error if you don't have your environment set to use `C.UTF-8` as the `LANG`.  Others, such as `en_US.UTF-8` or `UTF-8` haven't worked for me. 

As of commit https://github.com/viper-framework/viper/commit/5cf700761d904d9002a272164735c2ed2b4e0df9 adding the following line to your Dockerfile will fix this issue.

`ENV LANG C.UTF-8` 


*This message added as a breadcrumb to save someone else from wasting the time I did trying to figure this out*
